### PR TITLE
Add forgotten doneInserting() after dynamic insertion of DistFCT chares

### PR DESCRIPTION
**Pull request by jbakosi**
I'm hoping this will be a fix for a problem that came out recently during a single regression test of about 500+ times running the full test suites in recent days which triggered the `Assert`:
```
"DistFCT ckLocal() null"
```
Considering we have 80 tests for inciter, that is a single `Assert` triggered out of roughly 40K inciter runs. I love asynchronous programming! ;-)

This is also why to write asserts generously, e.g., https://github.com/quinoacomputing/quinoa/pull/184#discussion_r162208076.

### CODE REVIEW

File: src/Inciter/Discretization.C [hunk 1]
```diff
@@ -127,9 +127,11 @@ Discretization::Discretization(
   // also, e.g., MatCG::next().
   const auto sch = g_inputdeck.get< tag::selected, tag::scheme >();
   const auto nprop = g_inputdeck.get< tag::component >().nprop();
-  if ((sch == ctr::SchemeType::MatCG || sch == ctr::SchemeType::DiagCG))
+  if ((sch == ctr::SchemeType::MatCG || sch == ctr::SchemeType::DiagCG)) {
     m_fct[ thisIndex ].insert( m_transporter, nchare, m_gid.size(), nprop,
                                m_msum, m_bid, m_lid, m_inpoel, CkMyPe() );
+    m_fct[ thisIndex ].doneInserting();
```
[the remark this answers is not in the document]
Comment (jbakosi) [reply src/Inciter/Discretization.C]: Yes, indeed. Nice catch! I forgot to switch this one to a broadcast.

**Comment by adityakpandare** [comment]
There seems to be a conflict here. I'm very interested to see how the resolution is done. Do you do it via the web UI or terminal?

**Comment by jbakosi** [comment]
Not through the web. This happens all the time because we are working on concurrent branches.

This time you pushed before me, before that I pushed before you and you did the right thing, and merged the current develop into your branch, but probably there were no conflicts. I usually just pull the feature branch, merge in (the new) develop) and git will (as github) tell you what files are conflicting. This case it is only a single file. Then I edit those files and fix the conflicts. These happen when the same line gets edited by both branches: develop and the the current feature branch. Usually a human knows what to do, since the computer/git cannot: same line was changed. So I do:
```
$ git checkout wait4fct
$ git merge develop
```
I just did that, I have no conflict. That's because I think github is trying to do what's behind its _command line instructions_. That is when you want to merge this feature into develop. I don't want that yet, because I want to just merge develop into wait4fct (the first half), then I will push, let travis build/test and when that is fine, I will merge, i.e., do the 2nd half of github's instructions.

(I think I will have that file conflicting at the 2nd step. You usually just have to edit the file and pick the right line depending which one you want. Git usually tells you what you need to do during `git status`.)

**Comment by jbakosi** [comment]
Okay, I forgot to pull your changes to my local `develop`. That is why it was not conflicting. Now when I checkout develop first, then pull, then checkout wait4fct, then merge develop into it, I get the conflict:
```
$ git checkout develop
$ git pull
$ git checkout wait4fct
$ git merge develop
Auto-merging src/Inciter/Transporter.h
Auto-merging src/Inciter/Transporter.C
Auto-merging src/Inciter/Partitioner.C
CONFLICT (content): Merge conflict in src/Inciter/Partitioner.C
Automatic merge failed; fix conflicts and then commit the result.
```
Now I edit `src/Inciter/Partitioner.C`, and find the problem:
```
<<<<<<< HEAD
      edno, m_nchare, m_nbfac, m_bface, m_belem, CkMyPe() );
=======
      edno, m_nchare, m_nbfac, m_bface, m_triinpoel, CkMyPe() );
    m_scheme.doneDiscInserting< tag::elem >( cid );
>>>>>>> develop
```
So I see that we both edited this, and we really want a combination of both your (develop) and my (HEAD) changes:
```
    m_scheme.discInsert< tag::elem >( cid, m_host, m_bc,
      tk::cref_find(m_chinpoel,cid), msum, tk::cref_find(m_chfilenodes,cid),
      edno, m_nchare, m_nbfac, m_bface, m_triinpoel, CkMyPe() );
    m_scheme.doneDiscInserting< tag::elem >( cid );
```

**Comment by adityakpandare** [comment]
Thanks for your answers to my questions about git conflicts. I can't see them here, but I do have the emails with your comments. I have always wondered how to resolve conflicts (and if possible how to avoid them completely)!

**Comment by jbakosi** [comment]
For some reason github deletes the comments when new commits are pushed to the PR. Not always, I'm sure there is a reason, e.g., when a comment relates to a line of code that gets changed or deleted. But in this case, a normal comment to a PR and answers? I have no idea what do to about it and how to avoid it. Best to avoid it, I guess.

**Pull request merged**